book_maker/utils.py

```
@@ -162,3 +167,195 @@ def num_tokens_from_text(text, model="gpt-3.5-turbo-0301"):
+                           Also may contain keys:
+                             - batch_size
+                             - language
+                             - model
+                             - context_flag
+                             - single_translate
+                             - api_calls
+                             - reasoning_effort
+                           as well as prompt file and message info.
+    """
+    import datetime
+    import os
```

Ruff (0.11.9)

Redefinition of unused `os` from line 4

Remove definition: `os`

(F811)